src/providers/incomplete-tests/incomplete-tests.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { SlotItem } from '../slot-selector/slot-item';
+import { SlotSelectorProvider } from '../slot-selector/slot-selector';
+import { TestStatus } from '../../modules/tests/test-status/test-status.model';
+import { AppConfigProvider } from '../app-config/app-config';
+
+@Injectable()
+export class IncompleteTestsProvider {
+
+  constructor(
+    public appConfig: AppConfigProvider,
+    private slotSelector: SlotSelectorProvider,
+  ) { }
+
+  countIncompleteTests = (testsStatuses: { [slotId: string]: TestStatus }, slots: SlotItem[]): number => {
+    if (!testsStatuses || !slots) {
+      return;
+    }
+
+    let count = 0;
+    slots.forEach((slot) => {
```

Comment:
I'd generally prefer `Array.prototype.reduce` for folding over a datastructure to do some counting such as this.